### ISSUE TITLE
Seems can`t block Sentinelone

### ISSUE THREAD

**Issue by Maskhe** [netero1010/EDRSilencer#20]
![image](https://github.com/user-attachments/assets/c3bff942-2dcf-43d5-8998-b41cebd0e114)

filter adds failed:

![image](https://github.com/user-attachments/assets/647ba0af-d578-4bef-991f-271a7811e8de)

**Comment by sec13b** [comment]
what are the process in task ?
~~~~~~~~~~~~~~~~~~~~~~
"SentinelOne\\", 
"C:\\Program Files\\SentinelOne", 
"SentinelAgent", 
"SentinelMonitor", 
~~~~~~~~~~~~~~~~~~~~~~

~~~~~~~~~~~~~~~~~~~~~~
// SentinelOne
    "SentinelAgent.exe",
    "SentinelAgentWorker.exe",
    "SentinelServiceHost.exe",
    "SentinelStaticEngine.exe",  
    "LogProcessorService.exe",
    "SentinelStaticEngineScanner.exe",
    "SentinelHelperService.exe",
    "SentinelBrowserNativeHost.exe",
~~~~~~~~~~~~~~~~~~~~~~

**Comment by Maskhe** [comment]
Sorry, I am not sure what exactly  you are asking。But I can explain my case again:
I use `EDRSilencer blockedr` to block sentinelone，even though the stdout information shows some wpf filters are created(pic 1)，but when I check the wpf state through command ` netsh wfp show state` ,I found these filters not created correctly.

**Comment by sec13b** [comment]
to block or to kill?

**Comment by Maskhe** [comment]
> to block or to kill?

block

**Comment by sec13b** [comment]
i think is enough to add : 

// SentinelOne
    "SentinelAgent.exe",
    "SentinelAgentWorker.exe",
    "SentinelServiceHost.exe",
    "SentinelStaticEngine.exe",  
    "LogProcessorService.exe",
    "SentinelStaticEngineScanner.exe",
    "SentinelHelperService.exe",
    "SentinelBrowserNativeHost.exe",


if you see another process , just paste here

**Comment by Frogsecurity** [comment]
Very curious about this as well. Does this resolve being able to block sentinelone outbound traffic? Those process names already seem to be present edrsilencer?

**Comment by Maskhe** [comment]
> Very curious about this as well. Does this resolve being able to block sentinelone outbound traffic? Those process names already seem to be present edrsilencer?

--------

The truth is no.

I am not skilled at WPF mechanism, but are there some configuration priorities in WPF? Because I observed that Sentinelone itself sets some wpf rules，maybe these rules make EDRSilencer invalid.

![image](https://github.com/user-attachments/assets/335e0abb-b360-4550-b7ec-2ed1c464f9e8)

![image](https://github.com/user-attachments/assets/0c503d82-d794-45b5-a0bf-e678b84050b5)

**Comment by sec13b** [comment]
can you see the sentinel wpf rules ?

**Comment by Maskhe** [comment]
> can you see the sentinel wpf rules ?

yes I can and you can also see some of them through the pictures I pasted in my previous comment

**Comment by sec13b** [comment]
i dont understand why sentinel install or run from \device\hardvolume3\ is like run from a restore system point ?

did you try any solution to kill ? and in the kill process to see another process who run outbound traffic from sentinel

**Comment by sec13b** [comment]
> > can you see the sentinel wpf rules ?
> 
> yes I can and you can also see some of them through the pictures I pasted in my previous comment

https://www.huntress.com/blog/silencing-the-edr-silencers

**Comment by eqvsec** [comment]
Please see Issue #21 that I just created. It should solve your problem.

**Comment by netero1010** [comment]
Thank you for @eqvsec 's help. I have updated the new commit 0e73a70. Please try the new code or new pre-compiled binary and see if it is fixed.

**Comment by eqvsec** [comment]
> Thank you for @eqvsec 's help. I have updated the new commit [0e73a70](https://github.com/netero1010/EDRSilencer/commit/0e73a7037ec65c52894d8208e6f605a7da0a34a6). Please try the new code or new pre-compiled binary and see if it is fixed.

This may be a new development. It seems now that the SentinelOne WFP sublayer has a higher weight than the sublayer EDRSilencer creates filters in. During sublayer arbitration, I believe this higher weight causes the SentinelOne sublayer/filters to have priority over the filters created in the sublayer EDRSilencer uses. Interestingly, it seems the sublayer weight used by SentinelOne is 65532 - not the maximum of 65535.

![image](https://github.com/user-attachments/assets/f11302dc-237e-4979-9642-4926e51bd0f7)

**Comment by netero1010** [comment]
Yes. In this case, the S1 filter will dominate useless a new sublayer is created with higher weight than S1. I will keep it in my to-do list if it is viable.

**Comment by netero1010** [comment]
> > Thank you for @eqvsec 's help. I have updated the new commit [0e73a70](https://github.com/netero1010/EDRSilencer/commit/0e73a7037ec65c52894d8208e6f605a7da0a34a6). Please try the new code or new pre-compiled binary and see if it is fixed.
> 
> This may be a new development. It seems now that the SentinelOne WFP sublayer has a higher weight than the sublayer EDRSilencer creates filters in. During sublayer arbitration, I believe this higher weight causes the SentinelOne sublayer/filters to have priority over the filters created in the sublayer EDRSilencer uses. Interestingly, it seems the sublayer weight used by SentinelOne is 65532 - not the maximum of 65535.
> 
> ![image](https://private-user-images.githubusercontent.com/132799545/383353036-f11302dc-237e-4979-9642-4926e51bd0f7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA4NTgxNzAsIm5iZiI6MTczMDg1Nzg3MCwicGF0aCI6Ii8xMzI3OTk1NDUvMzgzMzUzMDM2LWYxMTMwMmRjLTIzN2UtNDk3OS05NjQyLTQ5MjZlNTFiZDBmNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwNlQwMTUxMTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NDk4NGIxYzE2OTU1Y2I3NGUwNGU3NGU1OGEyYTkxMzZmY2JhMDI1ZDFhZmY5ZTgwNzNhZmVjOTI5Mjc3YWYyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.LweojWlVG36qBW01OzCwbJSUxkL7tnxF86VkuEx6FH4)


Yes. In this case, the S1 filter will dominate useless a new sublayer is created with higher weight than S1. I will keep it in my to-do list if it is viable.
